File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Madagascar.m
//
//  UIImage+Madagascar.m
//  CountriesFlags
//

#import "Countries/UIImage+Madagascar.h"

@implementation UIImage (Madagascar)

+ (UIImage * _Nullable)flagMadagascarWithSize:(CGSize)size {
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorWarmRed = [UIColor colorWithRed:249.0f/255.0f
                                            green:66.0f/255.0f
                                             blue:58.0f/255.0f
                                            alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:132.0f/255.0f
                                           blue:61.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorWarmRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 2.0f, size.width, size.height / 2.0f));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width / 3.0f, size.height));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Madagascar.swift
//
//  UIImage+Madagascar.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Madagascar "wikipedia"
     
     Generates the flag of Madagascar.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Madagascar.
     */
    internal static func flagMadagascar(with size: CGSize) -> UIImage? {
        let colorWhite = UIColor.white
        let colorWarmRed = UIColor(red: 249.0/255.0,
                                   green: 66.0/255.0,
                                   blue: 58.0/255.0,
                                   alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 132.0/255.0,
                                 blue: 61.0/255.0,
                                 alpha: 1.0)
        
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        
        colorWarmRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 2.0),
                          size: CGSize(width: size.width, height: size.height / 2.0)))
        
        colorWhite.setFill()
        UIRectFill(CGRect(origin: .zero,
                          size: CGSize(width: size.width / 3.0, height: size.height)))
        
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return image
    }
}

#endif
